### ISSUE TITLE
Support for variables in the header-filter regex during code analysis 

### ISSUE THREAD

**Issue by hriekehof** [microsoft/vscode-cpptools#9996]
### Feature Request

Currently i work in a large legacy code base which requires a lot of refactoring. I already can use  `Run Code Analysis on Active File` and i see the issues in the problems tab and also can apply fixes etc.
However i really would like to limit the headers which are included in the analysis based on the active file name for example. I created a task for that which looks like this:
```json
{
            "label": "Clang-Tidy on Active File",
            "type": "shell",
            "command": "/opt/homebrew/opt/llvm/bin/clang-tidy --config-file=/somewhere/.clang-tidy --header-filter=${fileBasenameNoExtension} -p ${workspaceFolder} -p ${file}",
            "group": "none",
            "problemMatcher": []
        }
```
By this i only get the analysis for the current cpp file and the corresponding header file. However with the task its just in the terminal and i don't have the advantages of the problems tab etc. It would be great to use some variables like `${fileBasenameNoExtension}` in the `headerFilter` setting.
Something like `"C_Cpp.codeAnalysis.clangTidy.headerFilter": "${fileBasenameNoExtension}",`

For me this would be great to be able to use fix all, for `misc-const-correctness` for example but limit the scope of the header files applied to it. This enables smaller PRs which are more limited in scope.

**Comment by sean-mcmanus** [comment]
Is the ${fileBasenameNoExtension} variable sufficient? Or were there others from the list: https://code.visualstudio.com/docs/editor/variables-reference ?

**Comment by hriekehof** [comment]
In the case i described `${fileBasenameNoExtension}` is already sufficient. But i think these will also come in handy at some point:
- `${workspaceFolder}`
- `${workspaceFolderBasename}`
- `${relativeFileDirname}`
- `${fileDirname}`

**Comment by sean-mcmanus** [comment]
@hriekehof Ok.

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3
